planner/core/rule_column_pruning.go

```diff
@@ -241,6 +241,25 @@ func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) erro
 				p.schema.Columns = append(p.schema.Columns[:i], p.schema.Columns[i+1:]...)
 			}
 		}
+		// It's possible that the child operator adds extra columns to the schema.
+		// Currently, (*LogicalAggregation).PruneColumns() might do this.
+		// But we don't need such columns, so we add an extra Projection to prune this column when this happened.
+		for i, child := range p.Children() {
+			if p.schema.Len() < child.Schema().Len() {
+				// Copy and use the Union's schema as the Projection's Exprs and schema.
+				newCols := make([]expression.Column, len(p.schema.Columns))
+				exprs := make([]expression.Expression, len(p.schema.Columns))
+				for j, col := range p.schema.Columns {
+					newCols[j] = *col
```

Comment:
We don't need this change?

Reply:
Updated. Now the `Projection`'s schema and `Exprs` share the same `Column`s.